test/uk/gov/hmrc/tai/controllers/income/IncomeControllerSpec.scala

```diff
@@ -191,15 +200,452 @@ class IncomeControllerSpec extends PlaySpec
+            |             "payrollNumber":"888",
+            |             "startDate":"2019-05-26",
+            |             "annualAccounts":[
+            |
+            |             ],
+            |             "taxDistrictNumber":"",
+            |             "payeNumber":"",
+            |             "sequenceNumber":1,
+            |             "cessationPay":100,
+            |             "hasPayrolledBenefit":false,
+            |             "receivingOccupationalPension":true
+            |          }
+            |       }
+            |    ],
+            |    "links":[
+            |
+            |    ]
+            | }
+          """.stripMargin)
+
+      contentAsJson(result) mustBe expectedJson
+    }
+
+    "return empty json when there are no matching live employments" in {
+      val mockIncomeService = mock[IncomeService]
+      when(mockIncomeService.taxCodeIncomes(any(), Matchers.eq(TaxYear().next))(any()))
+        .thenReturn(Future.successful(taxCodeIncomes))
+
+      val mockEmploymentService = mock[EmploymentService]
+      when(mockEmploymentService.employments(any[Nino], any[TaxYear])(any[HeaderCarrier]))
+        .thenReturn(Future.successful(Seq(employment.copy(sequenceNumber = 99))))
+
+      val sut = createSUT(incomeService = mockIncomeService, employmentService = mockEmploymentService, authentication = loggedInAuthenticationPredicate)
+      val result = sut.matchedTaxCodeIncomesForYear(nino, TaxYear().next, "EmploymentIncome", "Live")(FakeRequest())
+
+      status(result) mustBe OK
+      contentAsJson(result) mustBe expectedJsonEmpty
+    }
+
+    "return empty json when there are no matching live pensions" in {
+      val mockIncomeService = mock[IncomeService]
+      when(mockIncomeService.taxCodeIncomes(any(), Matchers.eq(TaxYear().next))(any()))
+        .thenReturn(Future.successful(taxCodeIncomes))
+
+      val mockEmploymentService = mock[EmploymentService]
+      when(mockEmploymentService.employments(any[Nino], any[TaxYear])(any[HeaderCarrier]))
+        .thenReturn(Future.successful(Seq(employment.copy(sequenceNumber = 99))))
+
+      val sut = createSUT(incomeService = mockIncomeService, employmentService = mockEmploymentService, authentication = loggedInAuthenticationPredicate)
+      val result = sut.matchedTaxCodeIncomesForYear(nino, TaxYear().next, "PensionIncome", "Live")(FakeRequest())
+
+      status(result) mustBe OK
+      contentAsJson(result) mustBe expectedJsonEmpty
+    }
+
+    "return empty json when there are no TaxCodeIncome records for a given nino" in {
+      val mockIncomeService = mock[IncomeService]
+      when(mockIncomeService.taxCodeIncomes(any(), Matchers.eq(TaxYear().next))(any()))
+        .thenReturn(Future.successful(Seq.empty[TaxCodeIncome]))
+
+      val mockEmploymentService = mock[EmploymentService]
+      when(mockEmploymentService.employments(any[Nino], any[TaxYear])(any[HeaderCarrier]))
+        .thenReturn(Future.successful(Seq(employment)))
+
```

Comment:
This test looks really similar to the one above, is the failing case for the previous test also because of no given nino? It is difficult to see what the importance of sequenceNumber and nino is with any matchers

Reply:
not sure which test you're referring too here since line numbers have changed, in terms of importance of seq number, that's what the matching is done on in the business logic that we have moved from what was in TAI FE to TAI BE. Why this actually is used for the matching is not something we could answer, it's just what is being done for some business reason, regardless, it belongs in the backend.